h/static/scripts/login-forms/components/SignupForm.tsx

```diff
@@ -40,11 +82,38 @@ export default function SignupForm() {
   );
   const commsOptIn = useFormValue(config.formData?.comms_opt_in ?? false);
 
+  useLayoutEffect(() => {
+    const url = new URL(location_.href);
+    const idInfo = url.searchParams.get('idinfo');
+    if (!idInfo) {
+      return;
+    }
+    setIdInfoJWT(idInfo);
+  }, [location_]);
```

Comment:
`useLayoutEffect` is a confusing name. It's essentially the same as `useEffect` but it runs synchronously after the first render, whereas `useEffect` runs asynchronously.